src/scalarstats.jl

```diff
@@ -158,6 +158,49 @@ function modes(a)
     return [x for (x, c) in cnts if c == mc]
 end
 
+# Weighted mode of arbitrary vectors of values
+function mode(a::AbstractVector, wv::AbstractWeights{T}) where T <: Real
+    isempty(a) && throw(ArgumentError("mode is not defined for empty collections"))
+    length(a) == length(wv) || throw(ArgumentError(
+        "data and weight vectors must be the same size, got $(length(a)) and $(length(wv))"
+    ))
```

Comment:
```suggestion
    length(a) == length(wv) ||
        throw(ArgumentError("data and weight vectors must be the same size, got $(length(a)) and $(length(wv))"))
```

Reply:
This wasn't actually applied.

Reply:
Oh, no it was, but then I forgot that I applied it on github, so my rebase with master lossed it.